File: main/java/ru/carwash/screens/main/orders/CarsInOrderAdapter.java
package ru.carwash.screens.main.orders;

import android.content.Context;
import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.fragment.app.FragmentActivity;
import androidx.recyclerview.widget.RecyclerView;

import com.carwash.carwash.R;

import java.util.ArrayList;

import ru.carwash.dto.Car;

/* Адаптер для RecycleView списка машин, который находится во фрагменте создания заказа */
public class CarsInOrderAdapter extends RecyclerView.Adapter<CarsInOrderAdapter.ViewHolder> {

    private Context context; // для доступа к FragmentManager
    private ArrayList<Car> cars;

    private int selectedCar = RecyclerView.NO_POSITION;

    public CarsInOrderAdapter(FragmentActivity activity, ArrayList<Car> cars) {
        this.context = activity;
        this.cars = cars;
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {

        private final ConstraintLayout parent;
        private final TextView tvCarName;
        private final TextView imgCarNumber;
        private final TextView imgRegion;

        public ViewHolder(View view) {
            super(view);
            // Define click listener for the ViewHolder's View
            parent = view.findViewById(R.id.clCarItem);
            tvCarName = view.findViewById(R.id.tvCarName);
            imgCarNumber = view.findViewById(R.id.imgNumber);
            imgRegion = view.findViewById(R.id.imgRegion);
        }

        public ConstraintLayout getParent() { return parent; }
        public TextView getTvCarName() { return tvCarName; }
        public TextView getImgCarNumber() { return imgCarNumber; }
        public TextView getImgRegion() { return imgRegion; }
    }

    @NonNull
    @Override
    public CarsInOrderAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.car_in_order_item, parent, false);
        return new CarsInOrderAdapter.ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull CarsInOrderAdapter.ViewHolder holder, int position) {
        holder.itemView.setSelected(selectedCar == position);

        // изменяем цвет текста названия машины, если машина подсвечена/не подсвечена
        if(selectedCar == position) holder.getTvCarName().setTextColor(Color.WHITE);
        else holder.getTvCarName().setTextColor(Color.BLACK);

        holder.getParent().setOnClickListener(v -> { // обработчик нажатия на элемент в списке для подсветки выделенного авто
            notifyItemChanged(selectedCar);
            selectedCar = holder.getLayoutPosition();
            notifyItemChanged(selectedCar);
        });
        holder.getTvCarName().setText(cars.get(position).getBrand() + " " + cars.get(position).getModel());
        holder.getImgCarNumber().setText(cars.get(position).getCarNumber());
        holder.getImgRegion().setText(cars.get(position).getRegion());
    }



    @Override
    public int getItemCount() {
        return cars.size();
    }
}

File: main/java/ru/carwash/ui/main/orders/CarsInOrderAdapter.kt
package ru.carwash.ui.main.orders

import android.content.Context
import android.graphics.Color
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.fragment.app.FragmentActivity
import androidx.lifecycle.MutableLiveData
import androidx.recyclerview.widget.RecyclerView
import com.carwash.carwash.R
import ru.carwash.dto.Car
import java.util.*

/* Адаптер для RecycleView списка машин, который находится во фрагменте создания заказа */
class CarsInOrderAdapter(
        activity: FragmentActivity,
        cars: ArrayList<Car>,
        selectedCar: MutableLiveData<Car>) : RecyclerView.Adapter<CarsInOrderAdapter.ViewHolder>() {

    private val context // для доступа к FragmentManager
            : Context
    private val cars: ArrayList<Car>
    private var selectedCarIndex = RecyclerView.NO_POSITION
    private val selectedCar by lazy {
        selectedCar
    }

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val parent: ConstraintLayout = view.findViewById(R.id.clCarItem)
        val tvCarName: TextView = view.findViewById(R.id.tvCarName)
        val imgCarNumber: TextView = view.findViewById(R.id.imgNumber)
        val imgRegion: TextView = view.findViewById(R.id.imgRegion)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.car_in_order_item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.itemView.isSelected = selectedCarIndex == position

        // изменяем цвет текста названия машины, если машина подсвечена/не подсвечена
        if (selectedCarIndex == position) holder.tvCarName.setTextColor(Color.WHITE) else holder.tvCarName.setTextColor(Color.BLACK)
        holder.parent.setOnClickListener { v: View? ->  // обработчик нажатия на элемент в списке для подсветки выделенного авто
            notifyItemChanged(selectedCarIndex)
            selectedCarIndex = holder.layoutPosition
            notifyItemChanged(selectedCarIndex)
            selectedCar.value = cars[selectedCarIndex]
        }
        holder.tvCarName.text = cars[position].brand + " " + cars[position].model
        holder.imgCarNumber.text = cars[position].carNumber
        holder.imgRegion.text = cars[position].region
    }

    override fun getItemCount(): Int {
        return cars.size
    }

    init {
        context = activity
        this.cars = cars
    }
}